moveit_ros/planning/robot_model_loader/src/robot_model_loader.cpp

```diff
@@ -131,89 +131,78 @@ void RobotModelLoader::configure(const Options& opt)
+          bool has_acc_limits = false;
           if (!node_->has_parameter(param_name))
           {
-            has_acc_limits = node_->declare_parameter(param_name, false);
+            node_->declare_parameter(param_name, false);
           }
           node_->get_parameter(param_name, has_acc_limits);
 
-          // All types of joints can handle velocity and accletation limits
+          // All types of joints can handle velocity and acceleration limits
           joint_limit[joint_id].has_velocity_limits = has_vel_limits;
           joint_limit[joint_id].has_acceleration_limits = has_acc_limits;
 
           if (has_position_limits && canSpecifyPosition(joint_model, joint_id))
           {
-            double max_position = 0.0;
-            double min_position = 0.0;
-
             param_name = prefix + "max_position";
             if (!node_->has_parameter(param_name))
             {
-              max_position = node_->declare_parameter(param_name, 0.0);
+              node_->declare_parameter(param_name, 0.0);
             }
-            node_->get_parameter(param_name, max_position);
+            node_->get_parameter(param_name, joint_limit[joint_id].max_position);
 
             param_name = prefix + "min_position";
             if (!node_->has_parameter(param_name))
             {
-              min_position = node_->declare_parameter(param_name, 0.0);
+              node_->declare_parameter(param_name, 0.0);
             }
-            node_->get_parameter(param_name, min_position);
+            node_->get_parameter(param_name, joint_limit[joint_id].min_position);
 
             joint_limit[joint_id].has_position_limits = has_position_limits;
-            joint_limit[joint_id].max_position = max_position;
-            joint_limit[joint_id].min_position = min_position;
           }
 
           if (has_vel_limits)
           {
             param_name = prefix + "max_velocity";
-            double max_velocity = 0.0;
-
             if (!node_->has_parameter(param_name))
             {
-              max_velocity = node_->declare_parameter(param_name, 0.0);
+              node_->declare_parameter(param_name, 0.0);
             }
 
-            node_->get_parameter(param_name, max_velocity);
-            joint_limit[joint_id].max_velocity = max_velocity;
+            node_->get_parameter(param_name, joint_limit[joint_id].max_velocity);
           }
 
           if (has_acc_limits)
           {
             param_name = prefix + "max_acceleration";
-            double max_acc = 0.0;
 
             if (!node_->has_parameter(param_name))
             {
-              max_acc = node_->declare_parameter(param_name, 0.0);
+              node_->declare_parameter(param_name, 0.0);
             }
 
-            node_->get_parameter(param_name, max_acc);
-            joint_limit[joint_id].max_acceleration = max_acc;
+            node_->get_parameter(param_name, joint_limit[joint_id].max_acceleration);
```

Comment:
Sorry to nitpick but what if the user puts:

`has_acceleration_limit = true;`

But does not include `max_acceleration` in the yaml file?

I think we should print an error message:

`RCLCP_ERROR(logger, "has_acceleration_limits was set, but max_acceleration was not");`

Or just delete `has_vel/accel_limits` altogether?